pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -1188,6 +1208,14 @@ func (vs *VSphere) NodeDeleted(obj interface{}) {
 	}
 
 	glog.V(4).Infof("Node deleted: %+v", node)
+	disks, err := vs.nodeManager.GetAttachedDisks(convertToK8sType(node.ObjectMeta.Name))
+	if err == nil {
+		for _, disk := range disks {
+			if err := vs.DetachDisk(disk, convertToK8sType(node.ObjectMeta.Name)); err != nil {
```

Comment:
DetachDisk won't be retried if it fails (example: In case of network issue to vCenter) and still the volume will be attached to power off node. When kubernetes will retry to detach disk from the deleted node it will fail since node will be unregistered here.